Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -124,12 +124,23 @@ protected override void Dispose(bool disposing)
 				{
 					if (!activity.IsDestroyed)
 					{
-						FragmentManager fm = FragmentManager;
-						FragmentTransaction trans = fm.BeginTransaction();
-						foreach (Fragment fragment in _fragmentStack)
+						FragmentTransaction trans = FragmentManager.BeginTransaction();
+						trans.DisallowAddToBackStack();
+
+						while (_fragmentStack.Count != 0)
+						{
+							Fragment fragment = _fragmentStack.Last();
+							_fragmentStack.Remove(fragment);
```

Comment:
Removing `List` references to fragments. Not sure if this has any affect, but it somewhat bothered me that `Dispose` didn't let go of the list elements.